src/integrationTest/java/org/opensearch/security/FlsAndFieldMaskingTests.java

```diff
@@ -811,4 +847,287 @@ public void getFieldCapabilities() throws IOException {
         }
     }
 
+    @Test
+    public void testGetDocumentWithFLSRestrictions() throws Exception {
```

Comment:
Nit: Another naming thing; Can we make this illustrate what the test should do? For example testGetDocumentFLSNoTitles or something similar?